sympy/physics/control/lti.py

```diff
@@ -1422,3 +1445,551 @@ def _eval_rewrite_as_TransferFunction(self, num, den, **kwargs):
+                    temp.append(element._to_expr())
+            expr_mat_arg.append(temp)
+
+
+        if isinstance(arg, (list, Tuple)):
+            # Making nested Tuple (sympy.core.containers.Tuple) from nested list or nested python tuple
+            arg = Tuple(*(Tuple(*r, sympify=False) for r in arg), sympify=False)
+
+        obj = super(TransferFunctionMatrix, cls).__new__(cls, arg)
+        obj._expr_mat = ImmutableMatrix(expr_mat_arg)
+        return obj
+
+    @classmethod
+    def from_Matrix(cls, matrix, var):
+        """
+        Creates a new ``TransferFunctionMatrix`` efficiently from a SymPy Matrix of ``Expr`` objects.
+
+        Parameters
+        ==========
+
+        matrix : ``ImmutableMatrix`` having ``Expr``/``Number`` elements.
+        var : Symbol
+            Complex variable of the Laplace transform which will be used by the
+            all the ``TransferFunction`` objects in the ``TransferFunctionMatrix``.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import s, p, a
+        >>> from sympy.physics.control.lti import TransferFunctionMatrix
+        >>> from sympy.matrices import ImmutableMatrix
+        >>> from sympy.printing import pprint
+        >>> expr_1 = (s + 5)/(3*s**2 + 2*s + 1)
+        >>> expr_2 = 1/s
+        >>> expr_3 = 1/s**2
+        >>> expr_4 = 1/(s + 1)
+        >>> mat_1 = ImmutableMatrix([[expr_1, expr_2], [expr_3, expr_4]])
+        >>> mat_1
+        Matrix([
+        [(s + 5)/(3*s**2 + 2*s + 1),       1/s],
+        [                   s**(-2), 1/(s + 1)]])
+        >>> tfm_1 = TransferFunctionMatrix.from_Matrix(mat_1, s)
+        >>> tfm_1
+        TransferFunctionMatrix(((TransferFunction(s + 5, 3*s**2 + 2*s + 1, s), TransferFunction(1, s, s)), (TransferFunction(1, s**2, s), TransferFunction(1, s + 1, s))))
+        >>> pprint(tfm_1, use_unicode=False)
+        [    s + 5         1  ]
+        [--------------    -  ]
+        [   2              s  ]
+        [3*s  + 2*s + 1       ]
+        [                     ]
+        [      1           1  ]
+        [      --        -----]
+        [       2        s + 1]
+        [      s              ]
+        >>> expr_5 = 10
+        >>> expr_6 = 1/(s*(s - 1))
+        >>> expr_7 = p/(s - 1)
+        >>> expr_8 = a/(a*s + p)
+        >>> mat_2 = ImmutableMatrix([[expr_5, expr_6], [expr_7, expr_8]])
+        >>> mat_2
+        Matrix([
+        [       10, 1/(s*(s - 1))],
+        [p/(s - 1),   a/(a*s + p)]])
+        >>> tfm_2 = TransferFunctionMatrix.from_Matrix(mat_2, s)
+        >>> tfm_2
+        TransferFunctionMatrix(((TransferFunction(10, 1, s), TransferFunction(1, s*(s - 1), s)), (TransferFunction(p, s - 1, s), TransferFunction(a, a*s + p, s))))
+        >>> pprint(tfm_2, use_unicode=False)
+        [ 10        1    ]
+        [ --    ---------]
+        [ 1     s*(s - 1)]
+        [                ]
+        [  p        a    ]
+        [-----   ------- ]
+        [s - 1   a*s + p ]
+
+        """
```

Comment:
Is it really necessary to have so many examples and for them to be so long?

Note that these examples are for the documentation and are to be seen by users. If they do not convey information to users then they shouldn't be there.

I would have thought that this could just be one example:
```python
>>> M = Matrix([[s, 1/s], [1/(s+1), s]])
>>> Mtf = TransferFunctionMatrix(M, s)
>>> pprint(Mtf)
```
As far as I can tell the other examples don't really convey any meaningful information. Since the same as normal matrix printing the matrix is not really informative. Perhaps it would be better to briefly show some property that `Mtf` has that `M` does not like the poles or something.

Also why does this need to be `ImmutableMatrix`. Normal `Matrix` sympifies to `ImmutableMatrix` so if you use `sympify` then a normal `Matrix` could be used here.

Also Mathematica's version of this allows a matrix of numerators and a matrix of denominators. Is that supported here? Perhaps there could be a separate method for that.

Reply:
>Also Mathematica's version of this allows a matrix of numerators and a matrix of denominators. Is that supported here? Perhaps there could be a separate method for that.

The API has not been finalized yet, so I plan to add it in a separate PR later.